rst_files/mccall_model.rst

```diff
@@ -347,64 +319,36 @@ Our initial guess :math:`v` is the value of accepting at every given wage
 
 .. code:: julia
 
-    function plot_value_function_seq(plt,
-                                     c = 25,
-                                     β = 0.99,
-                                     num_plots = 6)
-
-        plt = deepcopy(plt)
-        v = collect(w_vals ./ (1 - β))
-        v_next = similar(v)
-
-        for i in 1:num_plots
-            plot!(plt, w_vals, v, label = "iterate $i")
-
-            # Update guess
-            for (j, w) in enumerate(w_vals)
-                stop_val = w / (1 - β)
-                cont_val = c + β * sum(v .* p_vals)
-                v_next[j] = max(stop_val, cont_val)
-            end
-            v[:] .= v_next
-        end
-        return plt
+    # parameters and constant objects
+    w_min, w_max = 10, 60
+    w_vals = range(w_min, w_max, length = n + 1)
+    c = 25
+    β = 0.99
+    num_plots = 6
+    E = expectation(dist)
+
+    # functions and matrix to plot
+    stop_val(w) = w / (1 - β)
+    cont_val(v) = c + β * E * v # using E as a linear operator
+    best_val(w, v) = max(stop_val(w), cont_val(v))
+    vs = repeat(w_vals ./ (1 - β), outer = (1, 6)) # data to fill
+
+    # fill and plot
+    for col in 2:num_plots
+        vs[:, col] .= best_val.(w_vals, Ref(vs[:, col - 1])) # update using previous value function
     end
-
-    plot_value_function_seq(plt)
-
+    plot(vs)
 
 Here's more serious iteration effort, that continues until measured deviation
 between successive iterates is below `tol`
 
-
 .. code-block:: julia
 
-    function compute_reservation_wage(c, β;
```

Comment:
I think it would be useful to resurrect this for teaching purposes.  Note that "The Algorithm" section, and (3) in particular, has this built in as an iteration.  We can then show the two different ways of doing it, where we show a particular, naive, fixed point algorithm and then call the library.  We don't need to do this for every time, but I think it would be helpful here.

Reply:
Added this (currently in a test block), pending what you want to do with it.